dot/core/service.go

```diff
@@ -304,8 +309,9 @@ func (s *Service) handleReceivedMessage(msg network.Message) (err error) {
 	return err
 }
 
-// checkForRuntimeChanges checks if changes to the runtime code have occurred; if so, load the new runtime
-func (s *Service) checkForRuntimeChanges() error {
+// HandleRuntimeChanges checks if changes to the runtime code have occurred; if so, load the new runtime
+// It also updates the BABE service and block verifier with the new runtime
+func (s *Service) HandleRuntimeChanges(header *types.Header) error {
```

Comment:
Does this need to be public?  It seems to only be called inside the service.

Reply:
the plan is for this to be called from the sync service as well once it imports a block (#1014) but could be changed back if that changes